src/zenml/zen_stores/schemas/artifact_schemas.py

```diff
@@ -184,13 +189,35 @@ def to_model(self) -> ArtifactResponseModel:
             uri=self.uri,
             materializer=materializer,
             data_type=data_type,
+            tags=json.loads(self.tags) if self.tags else None,
             created=self.created,
             updated=self.updated,
             producer_step_run_id=producer_step_run_id,
             metadata=metadata,
             visualizations=[vis.to_model() for vis in self.visualizations],
         )
 
+    def update(
+        self,
+        artifact_update: ArtifactUpdateModel,
+    ) -> "ArtifactSchema":
+        """Update an `ArtifactSchema` with an `ArtifactUpdateModel`.
+
+        Args:
+            artifact_update: The update model to apply.
+
+        Returns:
+            The updated `ArtifactSchema`.
+        """
+        if artifact_update.name:
+            self.name = artifact_update.name
+        if artifact_update.version:
+            self.version = str(artifact_update.version)
```

Comment:
To continue on the previous comment: I can think of situations where we want to change the name, but why users should be able to change versions? I consider version an immutable entity.

Reply:
Oh yes, I agree, version should be immutable 👍